web/modules/custom/dpl_redia_legacy/src/Controller/RssFeeds/EventsController.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+namespace Drupal\dpl_redia_legacy\Controller\RssFeeds;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\File\FileUrlGeneratorInterface;
+use Drupal\Core\Url;
+use Drupal\file\FileInterface;
+use Drupal\image\ImageStyleInterface;
+use Drupal\media\MediaInterface;
+use Drupal\recurring_events\Entity\EventInstance;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use function Safe\filesize;
+use function Safe\getimagesize;
+
+/**
+ * Building a Redia-legacy RSS feed, showing eventinstances.
+ */
+class EventsController extends ControllerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    protected FileUrlGeneratorInterface $fileUrlGenerator,
+    protected RequestStack $requestStack,
+  ) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): static {
+    return new static(
+      $container->get('file_url_generator'),
+      $container->get('request_stack'),
+
+    );
+  }
+
+  /**
+   * Getting the RSS/XML feed of the items.
+   */
+  public function getFeed(): Response {
+    $items = $this->getItems();
+
+    $rss_content = $this->buildRss($items);
+
+    $response = new Response();
+    $response->setContent($rss_content);
+    $response->headers->set('Content-Type', 'application/rss+xml');
+    return $response;
+  }
+
+  /**
+   * Loading events, and turning it into a simple array of relevant values.
+   *
+   * @return array<mixed>
+   *   An array of necessary item fields, used in buildRss().
+   */
+  private function getItems(): array {
+
+    $storage = $this->entityTypeManager()->getStorage('eventinstance');
+    $query = $storage->getQuery()
+      ->condition('status', TRUE)
+      ->accessCheck(TRUE)
+      ->sort('date.value');
+
+    $ids = $query->execute();
+
+    $events = EventInstance::loadMultiple($ids);
+
+    $items = [];
+
+    foreach ($events as $event) {
+      /** @var \Drupal\node\NodeInterface[] $branches */
+      $branches = $event->get('branch')->referencedEntities();
+      $branch = reset($branches);
+      $event_dates = $event->get('date')->getValue();
+      $changed_date = DrupalDateTime::createFromFormat('U', strval($event->getChangedTime()));
+
+      $items[] = [
+        'title' => $event->label(),
+        'description' => $this->getEventDescription($event),
+        'author' => $event->getOwner()->get('field_author_name')->getString(),
+        'id' => $event->id(),
+        'date' => $changed_date->format('r'),
+        'promoted' => FALSE,
+        'subtitle' => $event->get('event_description')->getString(),
```

Comment:
This is a bit confusing. Our event instances have a `description` field. This controller also has an `getEventDescription()` function.

I think I understand why this is needed but the current state of the code does not help me much.